### PR TITLE
[risk=no][RW-10271]  Ensure .R files show up on Analysis tab

### PR DESCRIPTION
**Pull request by NehaBroad**
https://github.com/all-of-us/workbench/assets/34481816/b17a89f8-6130-4c7f-b25b-481e62ef325b

https://github.com/all-of-us/workbench/assets/34481816/6dbe3067-b787-43cf-8d6d-29ac701448f0

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [x] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/NotebooksController.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ private FileDetail copyNotebookImpl(
       // needed.
       // TODO(yonghao): Remove withNotebookExtension after UI start setting extension.
       String newNameWithExtension =
-          isRstudioNotebook(fromNotebookNameWithExtension)
+          isRmdNotebook(fromNotebookNameWithExtension)
```
Comment (NehaBroad) [inline api/src/main/java/org/pmiops/workbench/api/NotebooksController.java:79]: Are Rstudio/R files consider notebooks or just files?

File: api/src/main/java/org/pmiops/workbench/notebooks/NotebookUtils.java [hunk 2]
```diff
@@ -17,22 +18,42 @@ private NotebookUtils() {}
...
   public static boolean isJupyterNotebook(String nameWithFileExtension) {
     return nameWithFileExtension.endsWith(JUPYTER_NOTEBOOK_EXTENSION);
   }
 
-  public static boolean isRstudioNotebook(String nameWithFileExtension) {
+  public static boolean isRmdNotebook(String nameWithFileExtension) {
     return nameWithFileExtension.endsWith(R_MARKDOWN_NOTEBOOK_EXTENSION);
   }
 
+  public static boolean isRStudioFiles(String nameWithFileExtension) {
```
Comment (evrii) [inline api/src/main/java/org/pmiops/workbench/notebooks/NotebookUtils.java:49]: Not a huge deal, but could you change this to `isRStudioFile`?

File: api/src/main/java/org/pmiops/workbench/notebooks/NotebooksServiceImpl.java [hunk 3]
```diff
@@ -342,7 +342,8 @@ public String getReadOnlyHtml(
   @Override
   public String adminGetReadOnlyHtml(
       String workspaceNamespace, String workspaceName, String notebookNameWithFileExtension) {
-    if (!NotebookUtils.isJupyterNotebook(notebookNameWithFileExtension)) {
+    if (!(NotebookUtils.isJupyterNotebook(notebookNameWithFileExtension)
```
Comment (evrii) [inline api/src/main/java/org/pmiops/workbench/notebooks/NotebooksServiceImpl.java:345]: Is this addition necessary? 

How I read this, it is saying:
If not a Jupyter notebook or if it is an RStudio file, throw a Not Implemented Exception.

I would suspect that that all RStudio files are not Jupyter notebooks.

File: api/src/main/java/org/pmiops/workbench/notebooks/NotebooksServiceImpl.java [hunk 4]
```diff
@@ -355,7 +356,9 @@ public String adminGetReadOnlyHtml(
             .getBucketName();
 
     Blob blob = getBlobWithSizeConstraint(bucketName, notebookNameWithFileExtension);
-    return convertJupyterNotebookToHtml(blob.getContent());
+    return NotebookUtils.isJupyterNotebook(notebookNameWithFileExtension)
```
Comment (evrii) [inline api/src/main/java/org/pmiops/workbench/notebooks/NotebooksServiceImpl.java:359]: If RStudio files throw an exception, is line 361 ever reached?

File: ui/src/app/pages/admin/admin-workspace.tsx [hunk 5]
```diff
@@ -168,8 +173,10 @@ const NameCell = (props: NameCellProps) => {
 
   // remove first check after RW-5626
   const isNotebook =
-    NOTEBOOKS_DIRECTORY === parseLocation(file, bucket) &&
-    filename.endsWith(NOTEBOOKS_SUFFIX);
+    (NOTEBOOKS_DIRECTORY === parseLocation(file, bucket) &&
```
Comment (evrii) [inline ui/src/app/pages/admin/admin-workspace.tsx:176]: This condition seem to say thatJupyter notebooks need to be in the notebooks directory, but R and Rmd files do not. Is that what we want?

File: api/src/test/java/org/pmiops/workbench/notebooks/NotebooksServiceTest.java [hunk 6]
```diff
@@ -710,4 +732,19 @@ public void testSaveNotebook_rmdNotSupported() {
             notebooksService.saveNotebook(
                 BUCKET_NAME, "test.Rmd", new JSONObject().put("who", "I'm a notebook!")));
   }
+
+  private void mockBlobsForHtml() {
```
Comment (evrii) [inline api/src/test/java/org/pmiops/workbench/notebooks/NotebooksServiceTest.java:736]: Thanks for making this utility function.

File: api/src/main/java/org/pmiops/workbench/notebooks/NotebooksServiceImpl.java (hunk 3 above)
Comment (evrii) [reply api/src/main/java/org/pmiops/workbench/notebooks/NotebooksServiceImpl.java:345]: Never mind, I misread where the parentheses were. This looks good!

**Review by evrii — APPROVED** [review]
Looks good to me. Thanks for talking this through with me.
